source/common/protobuf/utility.cc

```diff
@@ -614,48 +614,38 @@ using Transform = std::function<void(Protobuf::Message*, const Protobuf::Reflect
 bool redactOpaque(Protobuf::Message* message, bool ancestor_is_sensitive,
                   absl::string_view opaque_type_name, Transform unpack, Transform repack) {
   // Ensure this message has the opaque type we're expecting.
-  if (message->GetDescriptor()->full_name() != opaque_type_name) {
+  const auto* opaque_descriptor = message->GetDescriptor();
+  if (opaque_descriptor->full_name() != opaque_type_name) {
     return false;
   }
 
-  // Enumerate the fields of this opaque type. We expect to find exactly two: `type_url` and
-  // `value`, but not necessarily in that order.
-  const auto* reflection = message->GetReflection();
-  std::vector<const Protobuf::FieldDescriptor*> field_descriptors;
-  reflection->ListFields(*message, &field_descriptors);
-
-  Protobuf::DynamicMessageFactory message_factory;
-  std::unique_ptr<Protobuf::Message> typed_message;
-  const Protobuf::FieldDescriptor* value_field_descriptor{};
-  for (const auto* field_descriptor : field_descriptors) {
-    if (field_descriptor->name() == "type_url") {
-      // If we encounter the `type_url` field, try to find a descriptor for that type in the pool,
-      // and instantiate a new message using the `DynamicMessageFactory`.
-      std::string type_url(reflection->GetString(*message, field_descriptor));
-      std::string concrete_type_name(TypeUtil::typeUrlToDescriptorFullName(type_url));
-      const auto* descriptor =
-          Protobuf::DescriptorPool::generated_pool()->FindMessageTypeByName(concrete_type_name);
-      if (descriptor == nullptr) {
-        // If the type URL doesn't correspond to a known proto, give up redacting and treat this
-        // message the same as a `ProtobufWkt::Struct`. See the documented limitation on
-        // `MessageUtil::redact()` for more context.
-        ENVOY_LOG_MISC(warn, "Could not reify {} with unknown type URL {}", opaque_type_name,
-                       type_url);
-        return false;
-      }
-      typed_message.reset(message_factory.GetPrototype(descriptor)->New());
-    } else if (field_descriptor->name() == "value") {
-      // If we encounter the `value` field, hang on to its descriptor so that `unpack` and `pack`
-      // can use it to read and write.
-      value_field_descriptor = field_descriptor;
-    }
+  // Find descriptors for the `type_url` and `value` fields.
+  const auto* type_url_field_descriptor = opaque_descriptor->FindFieldByName("type_url");
+  const auto* value_field_descriptor = opaque_descriptor->FindFieldByName("value");
+  if (type_url_field_descriptor == nullptr || value_field_descriptor == nullptr) {
+    // In the unlikely event that the opaque type is malformed, don't try to reify it, just treat
```

Comment:
Is this even possible? I'm wondering if this should just be an assertion. I assume there are no tests, so this would be uncovered code.